### ISSUE TITLE
fails creation of dataset

### ISSUE THREAD

**Issue by travisghansen** [nmaupu/freenas-provisioner#4]
What version of FreeNAS are you on?  I'm currently running `11.1-U1` and I'm attempting to POST to create child dataset and it fails with 400 error (which doesn't print to the console by the way) with the following body:
```
{"name": ["Dataset names must begin with an alphanumeric character and may only contain \"-\", \"_\", \":\", \" \" and \".\"."]}
```
It seemingly doesn't like the fact that the `name` has a `/` in it.  Not sure how this is working for you unless something is off between version of the server.

**Comment by nmaupu** [comment]
I am using `FreeNAS-9.10.2-U6 (561f0d7a1)`

**Comment by travisghansen** [comment]
@nmaupu can you confirm if this endpoint works for you in 9.10: http://api.freenas.org/resources/storage.html#id2

**Comment by nmaupu** [comment]
This endpoint returns a 404 for me.
The following one is ok but quota (and comments) don't seem to work : http://api.freenas.org/resources/storage.html#datasets

Example:
```
curl -v -u root --http1.1 -X POST -H 'Content-Type: application/json' -d '{"name":"test", "quota":"100000"}' https://nas.home.fossar.net/api/v1.0/storage/volume/tank/datasets/
```

**Comment by travisghansen** [comment]
@nmaupu I just dropped a code dump in a MR.  Resolves all these issues for me.  I also found another fork that had some similar changes which I took into consideration.

**Comment by travisghansen** [comment]
Resolved by PR #6 . Closing.
